### ISSUE TITLE
tzdata update

### ISSUE THREAD

**Issue by dnovischi** [lildude/ha-addon-teslamate#45]
tzdata needs to be updated as it does not support any EET:

```
2024-09-25 11:33:40.909 [error] GenServer :tzdata_release_updater terminating
** (ArgumentError) errors were found at the given arguments:

  * 2nd argument: not a tuple

    :erlang.element(1, :error)
    (tzdata 1.1.1) lib/tzdata/util.ex:223: Tzdata.Util.to_int/1
    (tzdata 1.1.1) lib/tzdata/parser.ex:38: Tzdata.Parser.process_rule/1
    (tzdata 1.1.1) lib/tzdata/parser.ex:24: Tzdata.Parser.process_tz_list/1
    (tzdata 1.1.1) lib/tzdata/parser.ex:86: Tzdata.Parser.process_zone/5
    (tzdata 1.1.1) lib/tzdata/parser.ex:24: Tzdata.Parser.process_tz_list/1
    (tzdata 1.1.1) lib/tzdata/parser.ex:86: Tzdata.Parser.process_zone/5
    (tzdata 1.1.1) lib/tzdata/parser.ex:24: Tzdata.Parser.process_tz_list/1
Last message: :check_if_time_to_update

```

**Comment by lildude** [comment]
You to log this with the upstream TeslaMate project as this will be [this dependency](https://github.com/teslamate-org/teslamate/blob/6871ef73d40438abf80df07d9a543c510f79595f/mix.exs#L66) that needs updating.

**Comment by dnovischi** [comment]
Thanks. I'm all new to the teslamate / teslamate add-on in HA configuration. I have a fresh install with postgres, grafana and mqtt. Most of it is working, but many of the queries for grafana dashboards fail do to this issue.

Is there a way to update it myself? Or at least a time zone that will make the dashboards work, until the upstream fix.

**Comment by lildude** [comment]
According to [this upstream issue](https://github.com/teslamate-org/teslamate/issues/4196#issuecomment-2360905763), this should be fixed in the next release (this addon only uses releases and not master builds). You can't update it yourself but setting a time zone should do the trick, and when in doubt, always start with UTC. Try adding this to the `env_vars` addon configuration and restart the addon:

```yaml
- name: TZ
  value: UTC
```

**Comment by jer0enA** [comment]
Thanks @lildude but that still gives the same error.

**Comment by lildude** [comment]
It won't stop the error, but it should help with the graphs if I understand that issue correctly. If it doesn't you'll need to open an issue with the upstream project as something else is at play as all the graphs do is read from the Postgres DB.

**Comment by lildude** [comment]
I've just checked my HA and I see the errors but all my graphs work so something else is up with your env.

**Comment by lildude** [comment]
> Most of it is working, but many of the queries for grafana dashboards fail do to this issue.

What makes you believe this error is the cause? According to the issue, it's only a cosmetic issue.

What errors do you get from Grafana?

**Comment by dnovischi** [comment]
Hi,

I can config that it still behaves the same. There are many graphs which are not showing, for instance the "Odometer (From- To)", which of course fails since the actual query to postgres is:

```
Object
traceId:undefined
request:Object
url:"api/ds/query?ds_type=grafana-postgresql-datasource&requestId=Q140"
method:"POST"
data:Object
queries:Array[1]
from:"1726858680494"
to:"1727463480494"
hideFromInspector:false
response:Object
results:Object
A:Object
```


Another would be in the "Battery Healh" for Battery Capacity, which has the query:


```
Object
traceId:undefined
request:Object
url:"api/ds/query?ds_type=grafana-postgresql-datasource&requestId=Q153"
method:"POST"
data:Object
queries:Array[1]
from:"1727442078680"
to:"1727463678680"
hideFromInspector:false
response:Object
results:Object
A:Object
```


These are examples, but all querys done like this with from - to fail, for instance:  Ranges, Estimated Degradation, Battery Health, Current Stored Energy, Battery Capacity by Mileage, Drive Duration, Distance, Energy Used, Consumption, Speed Histogram, and I could go on and on, but I think you get the idea...

Simply put, dozens of graphs are rendered useless.

**Comment by dnovischi** [comment]
Also I have posted the issue to teslamate and the answer I've got is [this](https://github.com/teslamate-org/teslamate/issues/3040#issuecomment-2376583641).

Does the add-on use the master branch on which this issue is supposed to be solved?

**Comment by lildude** [comment]
I still don't see how this has anything to do with the tzdata error. The to-from values are time zone agnostic  and are part of the Grafana query to limit the range for the data in the DB. If you don't have enough data for the range or the DB doesn't contain the data, I'd expect certain graphs not to render properly or at all.

I suspect you may not have the data in the DB for some reason and I don't think it has anything to do with the tzdata error. You can confirm this by running the Grafana SQL query directly within the DB using either the psql cli command or a UI tool like pgadmin.

> Does the add-on use the master branch on which this issue is supposed to be solved?

No, as I said [before](https://github.com/lildude/ha-addon-teslamate/issues/45#issuecomment-2378469666):

> … (this addon only uses releases and not master builds) … 

Here's the line in the Dockerfile:

https://github.com/lildude/ha-addon-teslamate/blob/b9e37e9f321301dc9cc424a5d5b126316e5610c1/Dockerfile#L5

**Comment by dnovischi** [comment]
Thanks, I will look into it, though some of the errors from the ones mentioned before are like below:

![Screenshot from 2024-09-28 11-03-09](https://github.com/user-attachments/assets/05f9881e-a734-4c00-a491-5d5f140a81e9)

However, there may be other errors aside from tzdata one, as I also have things like this:

![Screenshot from 2024-09-28 11-09-13](https://github.com/user-attachments/assets/19e2e623-3df3-4a42-a10f-9feb0265546c)

**Comment by dnovischi** [comment]
Ok. I have played around with pgAdmin and the database seems fine. 
Also you were right about not accumulating some of the values
 
Otherwise those that depend on the time zone still fail. Plus it seems wired to me to have invalid drives ids since I have taken a quite few trips with the add-on in place, postgres, grafana and mqtt. 

I'll wait for the teslamate release since this seems to be the only option (for having timezone setting correctly picked up). By then I will probably have enough data for which all errors should be gone.
